R/utils.R

```diff
@@ -158,15 +158,22 @@ reset_lang <- function(old_lang) {
 #' @param fun A function that takes a source file and returns `lint` objects.
 #' @param name Default name of the Linter.
 #' Lints produced by the linter will be labelled with `name` by default.
+#' @param linter_level Which level of expression is the linter working with?
+#'   `"expression"` means an individual expression in `xml_parsed_content`, while `"file"` means all expressions
+#'   in the current file are available in `full_xml_parsed_content`.
+#'   `NA` means the linter will be run with both, expression-level and file-level source expressions.
+#'
 #' @return The same function with its class set to 'linter'.
 #' @export
-Linter <- function(fun, name = linter_auto_name()) { # nolint: object_name.
+Linter <- function(fun, name = linter_auto_name(), linter_level = c(NA_character_, "file", "expression")) { # nolint: object_name, line_length.
   if (!is.function(fun) || length(formals(args(fun))) != 1L) {
     stop("`fun` must be a function taking exactly one argument.", call. = FALSE)
   }
+  linter_level <- match.arg(linter_level)
```

Comment:
NB:

```r
match.arg(NA, NA_character_)
# Error in match.arg(NA, NA_character_) : 
#   'arg' must be NULL or a character vector
```

Maybe `match.arg(as.character(linter_level))` is better?

Reply:
Note that can omit `linter_level=` if you want `NA`. But yes, would be nicer to allow logical `NA` as well.